File: box-admin-web/src/api/plan.ts
import http, { type Result } from './http'

export interface PlanVO {
  id: number
  code: string
  name: string
  description?: string
  priceMonthly: number
  quotaAiCalls: number
  quotaTokens: number
  quotaMembers: number
  quotaWorkspaces: number
  quotaKnowledgeBases: number
  audience?: string
  overagePolicy?: string
  byokEnabled?: number
  status: number
  createdAt?: string
}

export interface CreatePlanRequest {
  code: string
  name: string
  description?: string
  priceMonthly: number
  quotaAiCalls: number
  quotaTokens: number
  quotaMembers: number
  quotaWorkspaces: number
  quotaKnowledgeBases: number
  audience?: string
  overagePolicy?: string
}

export interface UpdatePlanRequest {
  name: string
  description?: string
  priceMonthly: number
  quotaAiCalls: number
  quotaTokens: number
  quotaMembers: number
  quotaWorkspaces: number
  quotaKnowledgeBases: number
  audience?: string
  overagePolicy?: string
  status: number
}

export function fetchPlans() {
  return http.get<Result<PlanVO[]>>('/plans')
}

export function createPlan(payload: CreatePlanRequest) {
  return http.post<Result<PlanVO>>('/plans', payload)
}

export function updatePlan(id: number, payload: UpdatePlanRequest) {
  return http.put<Result<PlanVO>>(`/plans/${id}`, payload)
}

export function deletePlan(id: number) {
  return http.delete<Result<void>>(`/plans/${id}`)
}

File: box-server/box-domain/src/main/java/com/boxai/domain/plan/Plan.java
package com.boxai.domain.plan;

import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Getter
@Setter
public class Plan {

    private Long id;
    private String code;
    private String name;
    private String description;
    private BigDecimal priceMonthly;
    private Integer quotaAiCalls;
    private Long quotaTokens;
    private Integer quotaMembers;
    private Integer quotaWorkspaces;
    private Integer quotaKnowledgeBases;
    private String audience;
    private String overagePolicy;
    private Integer byokEnabled;
    private Integer status;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}
